Np dev (#51)
NiklasPfister/adaXT: README.md

## Fast Adaptable and Extendable Trees for Research

**adaXT** is a Python module for tree-based machine-learning

> algorithms that is fast, adaptable and extendable and aims to provide

Reviewer: ```suggestion
algorithms that is fast, adaptable and extendable and aims to provide
```

researchers a more flexible workflow when developing tree-based
models.

> It is distributed under the [3-Clause BSD license](LICENSE).

Reviewer: ```suggestion
It is distributed under the [3-Clause BSD license](https://github.com/NiklasPfister/adaXT/blob/main/LICENSE).
```
As the license doesn't have it's own page, you can't use the default referencing and you have
to leave a link to the site itself.

Author: Why is the full url not needed in the current version?

Reviewer: Not sure I follow. The current link doesn't work? If you do mkdocs serve, it will send you 
to a 404 page not found error.

We encourage users and developers to report problems, request
features, ask for help, or leave general comments.

Website: [https://NiklasPfister.github.io/adaXT](https://NiklasPfister.github.io/adaXT)

### Goals

The goal of adaXT is to provide a flexible and unified code-base for
various tree-based algorithms that strikes a balance between speed and
ease with which the code can be adapted and extended. It intends to
provide researchers a simple toolkit for prototyping new tree-based
algorithms.

adaXT aims to provide an intuitive user experience that is similar to
the [scikit-learn](https://scikit-learn.org) implementations of
decision trees both in terms model-based syntax and

> hyperparameter. Under the hood, however, adaXT stikes a different

Reviewer: ```suggestion
hyperparameter. Under the hood, however, adaXT strikes a different
```

balance between speed and ease of adapting and extending the code.

#### Adaptable and extendable

At the heart of any tree-based algorithm is a decision tree that can
be fitted on data and then used to perform some version of
prediction. adaXT has therefore been designed with a modular decision
tree implementation that takes four input components:

- Criterion class: Used during fitting to determine splits.

> - LeafBuilder class: Used during fitting to specify what is saved in

Reviewer: This should not be added until the given Prediction and LeafBuilder are on the main branch.
And preferably even in a new release.

Author: I would like to have this updated together with your proposed change that adds the leafbuilder and prediction class.

  the leaf nodes.

- Splitter class: Used during fitting to perform the splits.

- Prediction class: Used after fitting to make predictions.

> By specifying these three components a range of different tree

Reviewer: ```suggestion
By specifying these four components a range of different tree
```
I know that the splitter is the same for all, but stating four inputs, and then stating 
three components seems odd. If you want to keep it as three, perhaps add the specific names
e.g. Predict, Criteria and LeafBuilder.

algorithms can be created, e.g., regression trees, classification

> trees, quanitle regression trees and survial trees. Additionally to

Reviewer: ```suggestion
trees, quantile regression trees and survival trees. Additionally to
```

this modular structure, all other operations are kept as vanilla as
possible allowing users to easily change parts of the code (e.g., the
splitting procedure).

#### Speed

As tree-based algorithms involve evaluating expensive loops over the
dataset, it is important that these computations are implemented in a

> compiled language. adaXT implements all computationally expensive

Reviewer: The same about capitalization of AdaXT.

operations in [Cython](https://cython.org/). This results in speeds
similar (although a few factors slower) than the corresponding
[scikit-learn](https://scikit-learn.org) implementations. However, due

> to its modular structure and the avoidence of technical speed-ups

Reviewer: ```suggestion
to its modular structure and the avoidance of technical speed-ups
```
